bpf/cpu/cpu.bpf.c

```diff
@@ -321,22 +324,51 @@ static void bump_samples() {
 
 /*================================= EVENTS ==================================*/
 
+static __always_inline bool event_rate_limited(u64 event_id) {
+  if (!unwinder_config.event_rate_limits_enabled) {
+    return false;
+  }
+
+  int max_events_per_event_id = 2;
+
+  u32 one = 1;
+  u32* val = bpf_map_lookup_or_try_init(&events_count, &event_id, &one);
+  if (val) {
+    __sync_fetch_and_add(val, 1);
```

Comment:
Maybe I might be misremembering the details but this was failing on 5.4, no?

Reply:
I ran the vmtests here and it works fine. For some reason that I am not sure yet the code generated here vs in the interpreter unwinders tickles the verifier differently. We need to check the assembly to understand what's going on.

Reply:
(We also use atomically bump the stack_counts without issues)